cockatrice/src/server/chat_view/chat_view.cpp

```diff
@@ -457,6 +457,23 @@ bool ChatView::isModeratorSendingGlobal(QFlags<ServerInfo_User::UserLevelFlag> u
             (userLevel & ServerInfo_User::IsModerator || userLevel & ServerInfo_User::IsAdmin));
 }
 
+QString ChatView::getCurrentTime()
+{
+    if (showInGameTime) {
+        int seconds = *elapsedSeconds;
+        int minutes = seconds / 60;
+        seconds -= minutes * 60;
+        int hours = minutes / 60;
+        minutes -= hours * 60;
+        return QString("[%1:%2:%3] ")
```

Comment:
I think it might be easier to just use what qt provides here:
```suggestion
        return QTime::fromMSecsSinceStartOfDay(*elapsedSeconds * 1000).toString("[hh:mm:ss] ")
```
it's probably more performant to write all of this out ourselves, but using library code makes it easier to read and makes both code paths very similar, probably giving us an option to simplify this further.

Reply:
just noticed this is a copy from TabGame::incrementGameTime, itself some 14 year old code, maybe using qtime there and then having the resulting string be accessible in a getter for the chat view would be an option?